homeassistant/components/brunt/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""Config flow for brunt integration."""
+import logging
+
+from aiohttp import ClientResponseError
+from aiohttp.web import HTTPError
+from brunt import BruntClientAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class BruntConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for youless."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        bapi = BruntClientAsync(
+            username=user_input[CONF_USERNAME], password=user_input[CONF_PASSWORD]
+        )
+        try:
+            await bapi.async_login()
+        except (HTTPError, ClientResponseError):
+            _LOGGER.warning("Cannot connect to Brunt.")
+            return self.async_show_form(
+                step_id="user",
+                data_schema=DATA_SCHEMA,
+                errors={"base": "invalid_auth"},
```

Comment:
Same remark as above, is HTTPError always `invalid_auth`?  If you can catch the difference between internal server error (etc) and auth failed, it could help clarify the error messages.

But it will also allow you to trigger a reauth flow, which is handled via the interface.